src/org/zaproxy/zap/model/DefaultValueGenerator.java

```diff
@@ -0,0 +1,136 @@
+package org.zaproxy.zap.model;
+
+import java.net.URI;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+
+import net.htmlparser.jericho.FormControlType;
+
+public class DefaultValueGenerator implements ValueGenerator{
+	
+	private Date defaultDate;
+	private final String ATTR_TYPE = "type";
+	private final String DEFAULT_NUMBER_VALUE = "1";
+	private final String DEFAULT_TEXT_VALUE = org.parosproxy.paros.Constant.PROGRAM_NAME_SHORT;
+	private final String DEFAULT_PASS_VALUE = DEFAULT_TEXT_VALUE;
+	private final String DEFAULT_FILE_VALUE = "test_file.txt";
+	private final String DEFAULT_EMPTY_VALUE = "";
+	
+	/**
+	 * Gets the default {@code Date}, to be used for default values of date fields.
+	 *
+	 * @return the date, never {@code null}.
+	 * @see #setDefaultDate(Date)
+	 */
+	public Date getDefaultDate() {
+		if (defaultDate == null) {
+			return new Date();
+		}
+		return defaultDate;
+	}
+
+	/**
+	 * Sets the default {@code Date}, to be used for default values of date fields.
+	 * <p>
+	 * If none ({@code null}) is set it's used the current date when generating the values for the fields.
+	 *
+	 * @param date the default date, might be {@code null}.
+	 * @since TODO add version
+	 */
+	public void setDefaultDate(Date date) {
+		this.defaultDate = date;
+	}
+	
+
+	/**
+	 * <p>
+	 * Generates accurate field values for following types:
+	 * <ul>
+	 * <li>Text/Password/Search - DEFAULT_TEXT_VALUE</li>
+	 * <li>number/range - if min is defined, then use min, otherwise if max is defined use max
+	 * otherwise DEFAULT_NUMBER_VALUE;</li>
+	 * <li>url - http://www.example.com</li>
+	 * <li>email - contact@example.com</li>
+	 * <li>color - #000000</li>
+	 * <li>tel - 9999999999</li>
+	 * <li>date/datetime/time/month/week/datetime-local - current date in the proper format</li>
+	 * <li>file - DEFAULT_FILE_VALUE</li>
+	 * </ul> 
+	 */
+	
+	@Override
+	public String getValue(URI uri, Map<String, String> formAttributes, Map<String, String> fieldAttributes) {
+
+		if (fieldAttributes.get("Control Type") .equalsIgnoreCase (FormControlType.TEXT.toString())) {
+			//Converted FormControlType to String to allow for case insensitive comparison
+			
+			// If the control type was reduced to a TEXT type by the Jericho library, check the 
+			// HTML5 type and use proper values
+			String type = fieldAttributes.get(ATTR_TYPE);
+			
+			
+			if (type == null || type.equalsIgnoreCase("text")) {
+				return DEFAULT_TEXT_VALUE;
+			}
+
+			if (type.equalsIgnoreCase("number") || type.equalsIgnoreCase("range")) {
+				String min = fieldAttributes.get("min");
+				if (min != null) {
+					return min;
+				}
+				String max = fieldAttributes.get("max");
+				if (max != null) {
+					return max;
+				}
+				return DEFAULT_NUMBER_VALUE;
+			}
+
+			if (type.equalsIgnoreCase("url")) {
+				return "http://www.example.com";

```

Comment:
We do not believe there are constants defined to return these values for each HTML type. This is all from refactored code. Should we be creating constants to store these values?